test/sql/json/test_json_type.test

```diff
@@ -0,0 +1,155 @@
+# name: test/sql/json/test_json_type.test
+# description: Test JSON type
+# group: [json]
+
+require json
+
+statement ok
+pragma enable_verification
+
+# unary type function
+query T
+select json_type('{"str": 42}')
+----
+object
+
+query T
+select json_type('[1, 2, 3]')
+----
+array
+
+# this does not work as it is not valid JSON
+query T
+select json_type('"other"')
```

Comment:
Perhaps throwing an error is better in this case, e.g. in SQLite:

```sql
sqlite> select json_type('XXX');
Error: malformed JSON
```

Reply:
I think the end goal should be to implement all of [SQLite's json functions](https://www.sqlite.org/json1.html).

Reply:
I've implemented more of SQLite's json functions, still more to go